crawl-ref/source/dgn-overview.cc

```diff
@@ -1094,3 +1064,164 @@ bool connected_branch_can_exist(branch_type br)
 
     return true;
 }
+
+static int _prompt_annotate_branch(level_id lid)
+{
+    // XXX I'd be grateful if someone could refactor this into something that's
+    // not black magic. The functionality doesn't have the be the exact same,
+    // so long as the list is of reasonable length.
+    // these 3 lines make a vector containing all shown branches.
+    // the current implementation shows all known branches.
```

Comment:
the code for this is reasonbly readable, so i think this comment is probably not necessary